src/transaction.js

```diff
@@ -27,6 +30,10 @@ function Transaction(id, callback) {
       result.then(resolve, reject);
       return result;
     };
+
+    self._timeout = setTimeout(function() {
```

Comment:
Wdyt about keeping timeout as local variable only, clearing it in the `callback`.